src/adapters/MeetingsJSONAdapter.js

```diff
@@ -107,25 +121,29 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
       map((event) => event.detail)
     );
 
-    return concat(getMeetingWithLocalMedia$, muteEvent$);
+    return concat(getMeetingWithMedias$, muteEvent$);
   }
 
   /**
-   * Returns a MediaStream object obtained from local user media.
+   * Returns a video MediaStream object obtained from user local media.
+   * @param {MediaStreamConstraints} constraints  an object specifying the types of the media to request
    * @returns {MediaStream}
    */
-  async getLocalVideo() {
-    const constraints = {
-      video: true,
-      audio: false,
-    };
+  async getStream(constraints) {
     let stream;
 
     try {
-      stream = await navigator.mediaDevices.getUserMedia(constraints);
+      const mediaStream = await navigator.mediaDevices.getUserMedia(constraints);
+
+      // filter out either video or audio from a given constraints and return a new media stream
+      if (constraints.video) {
+        stream = new MediaStream([mediaStream.getVideoTracks()[0]]);
+      } else {
+        stream = new MediaStream([mediaStream.getAudioTracks()[0]]);
+      }
     } catch (reason) {
       // eslint-disable-next-line no-console
-      console.error('Meetings JSON adapter can not display the user local stream', reason);
+      console.error('Meetings JSON adapter can not display the user stream', reason);
```

Comment:
They're still local streams, why change the error message?

Reply:
I though it would make more more sense, since we are using it for the remote stream too. but I get your point.